src/nvim/eval.c

```diff
@@ -11682,8 +11682,18 @@ static void f_jobstart(typval_T *argvars, typval_T *rettv)
 
   bool pty = job_opts && get_dict_number(job_opts, (uint8_t *)"pty") != 0;
   bool detach = job_opts && get_dict_number(job_opts, (uint8_t *)"detach") != 0;
+  char *cwd = NULL;
+  if (job_opts) {
+    cwd = (char *)get_dict_string(job_opts, (char_u *)"cwd", false);
+    // The new cwd must be a directory.
+    if (cwd && !os_isdir((char_u *)cwd)) {
+      EMSG2(_(e_invarg2), "expected valid directory");
+      shell_free_argv(argv);
+      return;
```

Comment:
I fixed it by moving this above `common_job_callbacks()`. Ditto for the other comment.